subaction/matrix-targets/action.yml

```diff
@@ -0,0 +1,59 @@
+# https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions
+name: 'Bake matrix targets'
+description: 'Generate a matrix of targets to distribute builds in your workflow'
+
+inputs:
+  workdir:
+    description: Working directory
+    default: '.'
+    required: false
+  files:
+    description: Comma separated list of Bake files
+    required: false
+  target:
+    description: Bake target
+    default: 'default'
+    required: false
+
+outputs:
+  matrix:
+    description: Matrix of targets
+    value: ${{ steps.generate.outputs.matrix }}
+
+runs:
+  using: composite
+  steps:
+    -
+      name: Generate matrix
+      id: generate
+      uses: actions/github-script@v6
+      with:
+        script: |
+          let def;
+          const files = `${{ inputs.files }}` ? `${{ inputs.files }}`.split(',') : [];
+          const target = `${{ inputs.target }}` ? `${{ inputs.target }}` : 'default';
```

Comment:
I don't think this ternary is needed since we already set `default` as the default value for the input

```suggestion
```

Reply:
In the first implementation I was using the `.group` property to list targets so target name was necessary but after thought I switched to the `.target` property to list all keys instead. So yes it makes sense to remove the default.